File: python/n263_isUgly.py
# https://leetcode-cn.com/problems/ugly-number/submissions/
class Solution:
    def isUgly(self, num: int) -> bool:
        if num <= 0: return False
        while num > 1:
            if num % 2 == 0:
                num = num / 2
                continue
            if num % 3 == 0:
                num = num / 3
                continue
            if num % 5 == 0:
                num = num / 5
                continue
            else:
                return False
        return True


if __name__ == "__main__":
    pass

File: cpp/n263_isUgly.cpp
// https://leetcode-cn.com/problems/ugly-number/submissions/
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class Solution{
public:
    bool isUgly(int num){
        while (num % 2 == 0) num /= 2;
    }
}